### ISSUE TITLE
[AVM Module Issue]: module forces always managed identity creation

### ISSUE THREAD

**Issue by Hi-Fi** [Azure/terraform-azurerm-avm-res-containerregistry-registry#76]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

I'm not sure

### (Optional) Module Version

0.1.0 and 0.2.0

### Description

Currently there's no way to disable managed identity creation. Even setting `managed_identities.system_assigned` to `false`, System assigned identity is still created (as we don't want to assign user managed identity either).

For some reason with CDKTF setting `managedIdentities` to `null` doesn't create managed identities at all (as wanted), but with HCL version same thing doesn't seem to be working.

This doesn't create managed identities:
```json
  "module": {
    "AvmContainerRegistry": {
      "//": {
        "metadata": {
          "path": "test-stack/AvmContainerRegistry",
          "uniqueId": "AvmContainerRegistry"
        }
      },
      "location": "${azurerm_resource_group.ResourceGroup.location}",
      "managed_identities": null,
      "name": "testacr",
      "network_rule_bypass_option": "AzureServices",
      "public_network_access_enabled": false,
      "resource_group_name": "${azurerm_resource_group.ResourceGroup.name}",
      "source": "Azure/avm-res-containerregistry-registry/azurerm",
      "version": "0.2.0"
    }
  },
```

this does:
```HCL
module "AvmContainerRegistry" {
  location = "${azurerm_resource_group.ResourceGroup.location}"
  managed_identities = null
  name                          = "testacr"
  network_rule_bypass_option    = "AzureServices"
  public_network_access_enabled = false
  resource_group_name           = "${azurerm_resource_group.ResourceGroup.name}"
  source                        = "Azure/avm-res-containerregistry-registry/azurerm"
  version                       = "0.2.0"
}
```

#17 had something about this, but without any code references.

**Comment by jchancellor-ms** [comment]
@Hi-Fi - Thanks for reporting this.  I'm going to pick this up and include a fix in the next release.

**Comment by jchancellor-ms** [comment]
@Hi-Fi - If you leave the managed identities variable undefined it will default to an empty object which will not create the System managed identity or attach user managed identities.  However, setting is not allowed due to the logic in the managed_identities interface.

The default example doesn't set a managed identity and I've tested that it works properly in my environment.  Can you please try testing that leaving it undefined or setting it to a an empty object meets your needs?

`managed_identities = {}`

**Comment by Hi-Fi** [comment]
I'll try, I think I tried all options but that CDKTF `null` was only thing that didn't created the identity. I have to recheck this with HCL (as CDKTF has JSON formatted definition)

**Comment by Hi-Fi** [comment]
Seems that this is fixed with some version; when I created this bug only available version of module was 0.1.0, but with 0.3.1 it seems that identity is left uncreated as should.

Test file:
```hcl
provider "azurerm" {
  features {}
}

module "avm-res-containerregistry-registry" {
  source  = "Azure/avm-res-containerregistry-registry/azurerm"
  version = "0.3.1" # or 0.1.0
  location = "westeurope"
  name = "avmbugtest"
  resource_group_name = "rg-bug-test"
  # insert the 3 required variables here
}

```

With 0.1.0 (relevant part)
```
  # module.avm-res-containerregistry-registry.azurerm_container_registry.this will be created
  + resource "azurerm_container_registry" "this" {
      + identity {
          + principal_id = (known after apply)
          + tenant_id    = (known after apply)
          + type         = "SystemAssigned"
        }
    }
```

With 0.3.1
```
  # module.avm-res-containerregistry-registry.azurerm_container_registry.this will be created
  + resource "azurerm_container_registry" "this" {
      + admin_enabled                 = false
      + admin_password                = (sensitive value)
      + admin_username                = (known after apply)
      + anonymous_pull_enabled        = false
      + data_endpoint_enabled         = false
      + encryption                    = (known after apply)
      + export_policy_enabled         = true
      + id                            = (known after apply)
      + location                      = "westeurope"
      + login_server                  = (known after apply)
      + name                          = "avmbugtest"
      + network_rule_bypass_option    = "None"
      + network_rule_set              = (known after apply)
      + public_network_access_enabled = true
      + quarantine_policy_enabled     = false
      + resource_group_name           = "rg-bug-test"
      + retention_policy              = [
          + {
              + days    = 7
              + enabled = false
            },
        ]
      + sku                           = "Premium"
      + trust_policy                  = [
          + {
              + enabled = false
            },
        ]
      + zone_redundancy_enabled       = true
    }
```
